services/monitor-service/src/loadavg.rs

```diff
@@ -0,0 +1,216 @@
+//
+// Copyright (C) 2022 Xplore Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use std::f64;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::str;
+
+#[derive(Clone, Debug, Default, PartialEq, PartialOrd)]
+pub struct LoadAvg {
+    load_1m: Option<f64>,
+    load_5m: Option<f64>,
+    load_15m: Option<f64>,
+    processes_active: Option<u64>,
+    processes_total: Option<u64>,
+    last_pid: Option<u64>,
+}
+
+impl LoadAvg {
+    pub fn new() -> Self {
+        Self {
+            load_1m: None,
+            load_5m: None,
+            load_15m: None,
+            processes_active: None,
+            processes_total: None,
+            last_pid: None,
+        }
```

Comment:
just a minor suggestion -- since you already derive `Default`, and the implementation provides None for all `Option` types, you can use it to save quite a bit of work here..
```suggestion
        Default::default()
```

Reply:
That is some fantastic magic macros!